docs/Overview.md

```diff
@@ -0,0 +1,213 @@
+
+```javascript
+    var AwesomeView = React.createClass({
+      render: function () {
+        return (
+          <h1>Hello World!</h1>
+        );
+      }
+    });
+```
+
+## The Flux Architecture
+`TuxedoJS` is built on Facebook's [Flux](https://facebook.github.io/flux/) architecture. The two key concepts of `Flux` are one-way data flow and the `Dispatcher`. With one way data flow, views and `stores` (in `Flux` models/collections are replaced with stores) do not communicate back and forth as in standard MVC applications. Instead, views communicate with stores through a Dispatcher. This Dispatcher can be thought of as a Controller in normal MVC applications. However, unlike in standard MVC architectures, there is only one Dispatcher.
+
+The Dispatcher manages all the view and server events (in `Flux` these are called `actions`) and then dispatches those actions. Every store in a `Flux` application registers itself with the Dispatcher in order to listen to its dispatched actions and simply decides for itself which actions are relevant to it. Then, the stores that have updated emit update events to inform their views that they need to come and get the new data. This design dramatically simplifies data flow in an app and allows stores to fully encapsulate their data with as little coupling as possible. Normally, in a `Flux` application the Views will respond to user interaction by triggering `Action Helpers`, which usually will make Ajax requests or some other kind of asynchronous request and then dispatch the corresponding `Action` when the request completes.
+
+![Flux Architecture Diagram](http://facebook.github.io/flux/img/flux-simple-f8-diagram-with-client-action-1300w.png "Flux Architecture Diagram")
+
+## Where Tux Fits In
+`React` and `Flux` are both very powerful. They are even more powerful when used together since the two pieces fit together perfectly to provide all the functionality needed for an app. However, currently there is substantial glue code associated with getting `React` and `Flux` up and running and communicating with each other. More importantly, the standard `Flux` implementation relies on string comparisons and switch statements both of which are extremely fragile and can introduce difficult to track down bugs in a codebase. Lastly, although `Flux` decouples stores as much as possible, sometimes stores will still dependent on each other and managing those dependencies in `Flux` can be an absolute nightmare.
+
+`TuxedoJS` solves these problems by wrapping `React` and `Flux` together into a single framework that abstracts away all of the boilerplate and complexity associated with getting a `Flux` application up and running and communicating with React. Additionally, `Tux` Provides powerful augmented `React` Classes that are designed to handle different aspects of the app frontend such as managing state and listening to events from stores. The `TuxedoJS` app architecture looks like the following:
+
+NEED IMAGE
```

Comment:
If you don't think we're going to get this in there right now, remove it and add a chore for it.